src/unixfs-data-format.md

```diff
@@ -0,0 +1,652 @@
+  optional fixed32 FractionalNanoseconds = 2;
+}
+```
+
+Summarizing, a `dag-pb` UnixFS node is an IPLD [`dag-pb`][ipld-dag-pb] protobuf,
+whose `Data` field is a UnixFSV1 Protobuf message. For clarity, the specification
+document may represent these nested Protobufs as one object. In this representation,
+it is implied that the `PBNode.Data` field is encoded in a protobuf.
+
+## `dag-pb` Types
+
+A `dag-pb` UnixFS node supports different types, which are defined in
+`decode(PBNode.Data).Type`. Every type is handled differently.
+
+### `dag-pb` `File`
+
+A :dfn[File] is a container over an arbitrary sized amount of bytes. Files are either
+single block or multi-block. A multi-block file is a concatenation of multiple child files.
+
+#### The _sister-lists_ `PBNode.Links` and `decode(PBNode.Data).blocksizes`
+
+The _sister-lists_ are the key point of why IPLD `dag-pb` is important for files. They
+allow us to concatenate smaller files together.
+
+Linked files would be loaded recursively with the same process following a DFS
+(Depth-First-Search) order.
+
+Child nodes must be of type File; either a `dag-pb`:ref[File], or a
+[`raw` block](#raw-blocks).
+
+For example, consider this pseudo-json block:
+
+```json
+{
+  "Links": [{"Hash":"Qmfoo"}, {"Hash":"Qmbar"}],
+  "Data": {
+    "Type": "File",
+    "blocksizes": [20, 30]
+  }
+}
+```
+
+This indicates that this file is the concatenation of the `Qmfoo` and `Qmbar` files.
+
+When reading a file represented with `dag-pb`, the `blocksizes` array gives us the
+size in bytes of the partial file content present in children DAGs. Each index in
+`PBNode.Links` MUST have a corresponding chunk size stored at the same index
+in `decode(PBNode.Data).blocksizes`.
+
+:::warning
+Implementers need to be extra careful to ensure the values in `Data.blocksizes`
+are calculated by following the definition from [`Blocksize`](#decodepbnodedatablocksize).
+:::
+
+This allows for fast indexing into the file. For example, if someone is trying
+to read bytes 25 to 35, we can compute an offset list by summing all previous
+indexes in `blocksizes`, then do a search to find which indexes contain the
+range we are interested in.
+
+In the example above, the offset list would be `[0, 20]`. Thus, we know we only need to download `Qmbar` to get the range we are interested in.
+
+UnixFS parser MUST error if `blocksizes` or `Links` are not of the same length.
+
+#### `decode(PBNode.Data).Data`
+
+An array of bytes that is the file content and is appended before
+the links. This must be taken into account when doing offset calculations; that is,
+the length of `decode(PBNode.Data).Data` defines the value of the zeroth element
+of the offset list when computing offsets.
+
+#### `PBNode.Links[].Name`
+
+This field makes sense only in :ref[Directories] contexts and MUST be absent
+when creating a new file. For historical reasons, implementations parsing
+third-party data SHOULD accept empty values here.
```

Comment:
```suggestion
This field is only valid where in :ref[Directories] (i.e., where `Data.DataType`==1),
and MUST not be set when creating a new file. For historical reasons, implementations
parsing third-party data SHOULD accept empty values here.
```

Also, "For historical reasons" might be a little too vague for novices to understand
here; it might be more useful to say "data older than 20XX allowed this field to be 
empty, so parsers rejecting this to field being declared as empty will break 
compatibility when encountering historical data encoded before that deprecation.